### ISSUE TITLE
Problem using host names

### ISSUE THREAD

**Issue by joschu** [ray-project/ray#1176]
Summary: I tried using the host name of the head instead of the numerical IP, but the python ray client was unable to connect to it. It worked correctly if I used numerical IPs. Is this the intended behavior?

----

On the machine named `raytest-000`, I ran `ray start --head --node-ip-address=raytest-000 --redis-port=12345`, which seemed to work.
On another machine named `raytest-001`, I ran `ray start  --redis-address=raytest-000:12345`, which seemed to work.
On `raytest-001`, in python, I ran `ray.init(redis_address='raytest-000:12345')` which gives the error

    Some processes that the driver needs to connect to have not registered with Redis, so retrying. Have you run 'ray start' on this node?

I then ran the same sequence of commands using the numerical IPs (obtained from `/etc/hosts`, and it worked.

**Comment by robertnishihara** [comment]
We should probably make this work (brought up in #741), but I wouldn't expect it to work currently.

Let me give this a quick try, it might be a simple fix.

**Comment by joschu** [comment]
Different error this time:
```
In [1]: import ray

In [2]: ray.init(redis_address='raytest-000:12345')
Connection to IPC socket failed for pathname /tmp/plasma_manager29903103, retrying 50 times
---------------------------------------------------------------------------
ArrowIOError                              Traceback (most recent call last)
<ipython-input-2-36095ee297b0> in <module>()
----> 1 ray.init(redis_address='raytest-000:12345')

/opt/conda/lib/python3.6/site-packages/ray/worker.py in init(redis_address, node_ip_address, object_id_seed, num_workers, driver_mode, redirect_output, num_cpus, num_gpus, num_custom_resource, num_redis_shards, plasma_directory, huge_pages)
   1371                  num_redis_shards=num_redis_shards,
   1372                  plasma_directory=plasma_directory,
-> 1373                  huge_pages=huge_pages)
   1374
   1375

/opt/conda/lib/python3.6/site-packages/ray/worker.py in _init(address_info, start_ray_local, object_id_seed, num_workers, num_local_schedulers, object_store_memory, driver_mode, redirect_output, start_workers_from_local_scheduler, num_cpus, num_gpus, num_custom_resource, num_redis_shards, plasma_directory, huge_pages)
   1310             "webui_url": address_info["webui_url"]}
   1311     connect(driver_address_info, object_id_seed=object_id_seed,
-> 1312             mode=driver_mode, worker=global_worker, actor_id=NIL_ACTOR_ID)
   1313     return address_info
   1314

/opt/conda/lib/python3.6/site-packages/ray/worker.py in connect(info, object_id_seed, mode, worker, actor_id)
   1750     worker.plasma_client = plasma.connect(info["store_socket_name"],
   1751                                           info["manager_socket_name"],
-> 1752                                           64)
   1753     # Create the local scheduler client.
   1754     if worker.actor_id != NIL_ACTOR_ID:

/opt/conda/lib/python3.6/site-packages/ray/pyarrow_files/pyarrow/plasma.pyx in pyarrow.plasma.connect (/ray/src/thirdparty/arrow/python/build/temp.linux-x86_64-3.6/plasma.cxx:8059)()

pyarrow/error.pxi in pyarrow.lib.check_status (/ray/src/thirdparty/arrow/python/build/temp.linux-x86_64-3.6/lib.cxx:7951)()

ArrowIOError: Could not connect to socket /tmp/plasma_manager29903103

```

**Comment by robertnishihara** [comment]
Hmm.. can you try doing

```
git clean -f -f -x -d -e .whl
```

in the ray directory and then rerunning `python setup.py develop`?

My guess is that something about compilation didn't work when pulling the latest code.

Also, I'm guessing the same error happens if you just run

```python
```

**Comment by joschu** [comment]
You're right that ray.init failed. Will let you know when it's done compiling.

**Comment by joschu** [comment]
OK, recompiled. `ray.init()` now succeeds, but I get basically the same error as before.

**Comment by joschu** [comment]
And I made sure to do `pip uninstall ray` and `ray stop` on both machines.

**Comment by robertnishihara** [comment]
You're saying that `ray.init()` succeeds, but when you do

```python
ray.init(redis_address='raytest-000:12345')
```

you get the error

```
ArrowIOError: Could not connect to socket /tmp/plasma_manager29903103
```

is that right?

**Comment by joschu** [comment]
Correct. Also I just verified that everything still works when I put in the numerical IPs.

**Comment by robertnishihara** [comment]
Can you share the commands you're using to start Ray on the different machines.

**Comment by joschu** [comment]
on raytest-000: `ray start --head --node-ip-address=raytest-000 --redis-port=12345`
on raytest-001: `ray start --redis-address=raytest-000:12345`
on raytest-001, in python: `import ray; ray.init(redis_address='raytest-000:12345')`

**Comment by joschu** [comment]
It could be that there's something exotic about my network setup (on a kubernetes cluster) that's causing problems.

**Comment by robertnishihara** [comment]
Ok, I think I know what the problem is. Basically, I'm converting "hostname" -> "numerical IP address" for the `--redis-address`, but didn't do it for the `--node-ip-address`. Should have a fix soon.

**Comment by joschu** [comment]
Oh, looks like it's also working where I omit `--node-ip-address`, nice!

**Comment by robertnishihara** [comment]
Great! Should be fixed by the latest commit in #1177.

**Comment by robertnishihara** [comment]
Fixed by #1177.
